File: edu-platform.admin-master/edu-platform.admin-master/src/services/instructor/instructor.service.ts
import http from "./instructor.api";
import {InstructorGetListRequest} from "./request/InstructorGetListRequest";
import {InstructorGetListResponse} from "./response/InstructorGetListResponse";

const getInstructors = (request: InstructorGetListRequest) => {
  return http.get<InstructorGetListResponse>('/instructors', {
    params: request
  })
}

export const InstructorService = {
  getInstructors
}

File: edu-platform-master/edu-platform-master/src/main/java/edu/platform/domains/instructor/service/InstructorService.java
package edu.platform.domains.instructor.service;

import edu.platform.domains.instructor.model.request.*;
import edu.platform.domains.instructor.model.response.InstructorGetListResponse;
import edu.platform.domains.instructor.model.response.InstructorResponse;

public interface InstructorService {
    InstructorResponse save(InstructorSaveRequest request);

    InstructorResponse auth(InstructorAuthRequest request);

    InstructorResponse update(Long id, InstructorUpdateRequest request);

    void changePassword(Long id, InstructorChangePasswordRequest request);

    void active(Long id, InstructorActiveRequest request);

    InstructorGetListResponse getList(InstructorGetListRequest request);
}
